### PR TITLE
Fix actionlogs information emails containing HTML links

### PR DESCRIPTION
**Pull request by ManuelHu**
Pull Request for Issue #39851. This is essentially my strategy "3" as outlined in the issue, with the difference that the two replacement arrays are not merged. As I did not receive any comments on the issue, I just went ahead to implement this solution....

### Summary of Changes

This needed multiple changes:
1. `MailTemplate` now supports an extra set of replacement values for the plain-text mode. This will only be used if set explicitely, and will fall back to the old behaviour otherwise. The 'default' data and the plain data are _not_ merged. Callers have to ensure that both sets of values contain the same keys, or otherwise output might be missing.
1. `ActionlogModel` uses this new facility to provide values without HTML links.
1. `ActionlogsHelper::getHumanReadableLogMessage()` already had a parameter `$generateLinks`, that did not do what was documented. It only controlled the conversion from relative to absolute URLs. Now it will also remove link tags from the message templates, if set to false.

**Note**: This also affects the CSV export feature of the actionlogs components. Until now, the exported files contained relative urls in the administrator application (which is - I think - quite useless anyway). With this PR, links are also stripped. (This is the the only other call site that changed the value of `$generateLinks` to false.)

### Testing Instructions
* Install a recent 4.3.0 nighly build, that already has #39850 upmerged.
* Make sure action logs for user events are enabled and configure working system email sending
* In the user settings for a user with a valid email address (you can receive mail on), enable user action information emails for the user component.
* Log out and log back in
* Look at the emails you should have received

### Actual result BEFORE applying this Pull Request
Plain text emails contain HTML links

### Expected result AFTER applying this Pull Request
Plain text emails do not contain HTML links and only the link text.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/Mail/MailTemplate.php [hunk 1]
```diff
@@ -60,6 +60,13 @@ class MailTemplate
      */
     protected $data = [];
 
+     /**
```
Comment (Quy) [inline libraries/src/Mail/MailTemplate.php]:
```suggestion
    /**
```

**Comment by ManuelHu** [comment]
From a users perspective I would not classify this a s a new feature, but I also don't want to argue about that :-) (From the code point of view it is a new feature...)

To get this finally into 5.0, I would like to have some feedback if the slight b/c break here is accepatable:
* Adding a new parameter with a defult value should not even be a problem?
* The `$generateLinks` parameter of `ActionlogsHelper::getHumanReadableLogMessage()` changes it's semantics, but should not be covered by the b/c policy (as the function is part of a component helper, and not a core library)
* --> this has effects on the CSV export which now will also not contain links.

**Comment by ManuelHu** [comment]
I have rebased this to the current 5.0-dev. Can someone test this that we can get this timely for 5.0? @Quy

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on b8995aad151b6fa325fe7373c1d04657ddc50728<br /><br />Works fine: without patch email is hard to read because of links; with patch it is plain text.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40033">issues.joomla.org/tracker/joomla-cms/40033</a>.</sub>

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.1-dev.

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on b8995aad151b6fa325fe7373c1d04657ddc50728<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40033">issues.joomla.org/tracker/joomla-cms/40033</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40033">issues.joomla.org/tracker/joomla-cms/40033</a>.</sub>

**Comment by Razzo1987** [comment]
Thanks you!

**Pull request merged**